packages/amplify-graphql-relational-transformer/src/schema.ts

```diff
@@ -262,6 +268,13 @@ export const ensureHasManyConnectionField = (
   const relatedTypeObject = ctx.output.getType(relatedType.name.value) as ObjectTypeDefinitionNode;
   const connectionAttributeName = getConnectionAttributeName(ctx.featureFlags, object.name.value, field.name.value, connectionFieldName);
 
+  // The nullabilty of connection fields for hasMany depends on the other side belongsTo field
+  // Whereas in update input, they are always optional
+  const isConnectionFieldsNonNull = relatedTypeObject.fields?.some(field =>
+    getBaseType(field.type) === object.name.value
+    && field.directives?.some(d => d.name.value === 'belongsTo')
+    && isNonNullType(field.type)) ?? false;
```

Comment:
If the `@belongsTo` does not exist on the connected model, shouldn't we default to the nullability of the field, i.e `isNonNullType(field.type)` instead of `false`?

Reply:
I think this might need discussion on whether set the default to false and this is out of the scope of resolving the mentioned codegen issue. I will keep it as the original one as `isNonNullType(field.type)`